pkg/kubelet/kubelet_pods.go

```diff
@@ -619,6 +619,8 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod
 		return pod.Spec.ServiceAccountName, nil
 	case "status.podIP":
 		return podIP, nil
+	case "status.hostIP":
+		return kl.nodeIP.String(), nil
```

Comment:
I suggest to use [GetHostIP](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_getters.go#L218) instead.